### PR TITLE
Handle an s3 URI in asdf_cut()

### PR DESCRIPTION
**Pull request by snbianco**
If an s3 URI is passed in as the `input_file` parameter to the `asdf_cut` function, open the resource with `s3fs` and use its http URL in `asdf.open()`.

This prevents users from having to open the resource themselves before passing the file into the `asdf_cut` function.

### CODE REVIEW

**Comment by dr-rodriguez** [comment]
Double check if any of the unit tests need to be updated for this.

File: astrocut/asdf_cutouts.py [hunk 1]
```diff
@@ -247,8 +248,15 @@ def asdf_cut(input_file: str, ra: float, dec: float, cutout_size: int = 20,
         an image cutout object
     """
 
+    # if file comes from AWS cloud bucket, get URL
+    file = input_file
+    if isinstance(input_file, str) and input_file.startswith('s3://'):
```
Comment (havok2063) [inline astrocut/asdf_cutouts.py:253]: One thing we may want to do at some point is expand this to support `pathlib.Path` as input, in addition to a string.   I don't think `pathlib.Path` handles s3 urls, but it looks like there are packages that do, e.g. `s3path` or `s3pathlib`.   I don't think we need to do it for this PR, but maybe we can create an issue for the future.

**Review by havok2063 — COMMENTED** [review]
This looks good to me.  I would agree that we should add some explicit tests for string vs s3 path in `test_asdf_cut.py`.

File: astrocut/asdf_cutouts.py (hunk 1 above)
Comment (snbianco) [reply astrocut/asdf_cutouts.py:253]: Cool, will make a new issue for this!

**Comment by snbianco** [comment]
@dr-rodriguez @havok2063 

Trying to write unit tests for this now, and wondering what should be used for the `input_file` parameter since we don't have an s3 link with a working ASDF file yet. Also, should we be putting in a link to this data at all given that this is a public repo? I read on one of the other pull requests that the Roman test data shouldn't be publicly accessible.

Also, I'm not too familiar with `tox`, the testing library that we're using. Does anyone know how to generate a coverage report and/or where to find it?

**Comment by havok2063** [comment]
@snbianco We probably don't need to test a real file.  I think we can get away with a fake file for testing, or if there is a public ST S3 file to use for testing?  If we turn lines 251-257 into a tiny function, we could test that functionality separately from any asdf stuff.  That it returns the correct file or s3 file as input?  That would satisfy me for a test.    

It looks like there are packages for mocking s3 aws services, like `moto`.  I'm not sure if we need that here or not.

File: astrocut/asdf_cutouts.py (hunk 1 above)
Comment (snbianco) [reply astrocut/asdf_cutouts.py:269]: Issue is ASB-27085

**Comment by dr-rodriguez** [comment]
I generally recommend mocking external resources rather than relying on pings to public files. At a glance, it looked like your approach was good. Basically the goal is to confirm the functions were called with the parameters you expected; the exact logic of the s3fs functions is ideally something that package has verified already.

**Review by havok2063 — APPROVED** [review]
This looks good to me.

**Review by dr-rodriguez — APPROVED** [review]

**Pull request merged**